WinUIXamlCompiler/Transforms/XamlDirectPropertyResolver.cs

```diff
@@ -0,0 +1,45 @@
+using System.Collections.Generic;
+using System.Linq;
+using XamlX;
+using XamlX.Ast;
+using XamlX.Emit;
+using XamlX.IL;
+using XamlX.Transform;
+using XamlX.TypeSystem;
+using WinUIXamlCompiler.Ast;
+
+namespace WinUIXamlCompiler.Transforms
+{
+    class XamlDirectPropertyResolver : IXamlAstTransformer
+    {
+        public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode node)
+        {
+            if (node is XamlAstClrProperty prop)
+            {
+                if (prop.DeclaringType.Assembly == context.GetWinUITypes().WinUIControlsAssembly)
+                {
+                    IXamlField propertyIndexMaybe = context.GetWinUITypes().XamlPropertyIndex.Fields.FirstOrDefault(f => f.Name == $"{prop.DeclaringType.Name}_{prop.Name}");
+                    if (propertyIndexMaybe != null)
+                    {
+                        prop.Setters.Insert(0, new XamlDirectSetter(context.GetWinUITypes(), prop.Getter.ReturnType, prop.DeclaringType, propertyIndexMaybe));
+                        foreach (var adder in XamlTransformHelpers.FindPossibleAdders(context, prop.Getter.ReturnType))
```

Comment:
what does it mean to "FindPossibleAdders"?

Reply:
This finds add methods on the type of the property (like `UIElementCollection.Add`).